tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/AutoTuner.scala

```diff
@@ -0,0 +1,435 @@
+              val memoryOverhead = recommendMemoryOverhead(pinnedPoolSize,
+                bestConfig.get_executor_memory)
+              bestConfig.set_executor_memory_overhead(convertToHumanReadableSize(memoryOverhead))
+            }
+          }
+
+        case None =>
+          bestConfig.set_pinned_pool_size(DEFAULT_PINNED_POOL_SIZE)
+          val memoryOverhead = recommendMemoryOverhead(DEFAULT_PINNED_POOL_SIZE,
+            bestConfig.get_executor_memory)
+          bestConfig.set_executor_memory_overhead(convertToHumanReadableSize(memoryOverhead))
+      }
+    }
+  }
+
+  def getRecommendedProperties(): (Seq[RecommendedPropertyResult],
+    Seq[RecommendedCommentResult])= {
+    val systemProps = parseSystemInfo(workerInfo)
+    val recommendedConfig = new Config()
+    recommendSparkProperties(recommendedConfig, systemProps)
+    recommendGpuProperties(recommendedConfig, systemProps)
+    (toPropertyProfileResult(recommendedConfig), toCommentProfileResult(comments))
+  }
+}
+
+object AutoTuner extends Logging {
+  val SUPPORTED_UNITS = Seq("b", "k", "m", "g", "t", "p")
+
+  def parseSystemInfo(inputFile: String): SystemProps = {
+     try {
+       val yaml = new Yaml()
+       val file = scala.io.Source.fromFile(inputFile)
+       val text = file.mkString
+       val rawProps = yaml.load(text).asInstanceOf[java.util.Map[String, Any]]
+         .asScala.toMap.filter { case (_, v) => v != null }
+       val rawSystemProps = rawProps("system").asInstanceOf[java.util.Map[String, Any]]
+         .asScala.toMap.filter { case (_, v) => v != null }
+
+       if (rawSystemProps.nonEmpty) {
+         val rawGpuProps = rawProps("gpu").asInstanceOf[java.util.Map[String, Any]]
+           .asScala.toMap.filter { case (_, v) => v != null }
+
+         val gpuProps = if (rawGpuProps.nonEmpty) {
+           GpuProps(
+             rawGpuProps("count").toString.toInt,
+             rawGpuProps("memory").toString,
+             rawGpuProps("name").toString)
+         } else null
+
+         SystemProps(
+           rawSystemProps.getOrElse("num_cores", 1).toString.toInt,
+           rawSystemProps.getOrElse("cpu_arch", "").toString,
+           rawSystemProps.getOrElse("memory", "0b").toString,
+           rawSystemProps.getOrElse("free_disk_space", "0b").toString,
+           rawSystemProps.getOrElse("time_zone", "").toString,
+           rawSystemProps.get("num_workers").map(_.toString.toInt),
+           gpuProps)
+       } else null
+     } catch {
+       case e: Exception =>
+         logError("Exception: " + e.getStackTrace.mkString("Array(", ", ", ")"))
+         null
+     }
+  }
+
+  private def getSparkProperty(app: ApplicationSummaryInfo, property: String): Option[String] = {
+    app.sparkProps.collectFirst {
+      case propertyProfile
+        if propertyProfile.key == property && propertyProfile.rows(1) != "null" =>
+        propertyProfile.rows(1)
+    }
+  }
+
+  def convertFromHumanReadableSize(size: String): Long = {
```

Comment:
was this code copied from Spark?  If so having a comment pointing to it would be nice.  spark has some Utils class with some of this stuff, but its private so would have to use a trampoline to access.

Reply:
This method was created manually by me. I have added UTs for the same.